# Analyzing Conventional Random-Primed mRNA-Sequencing Data

This document introduces how to reproduce the analysis results of differentially expressed genes (DEGs) from the raw *FASTQ* data files generated by the conventional random-primed mRNA-sequencing method and deposited at *GEO*, by using the computational programs deposited at this GitHub repository.

## Software Requirements

The following software programs need to be installed for this data analysis:

- *Bash* shell in Unix-like operating systems (e.g. Linux, MacOS, Unix)
- *STAR* (version 2.5.3a)
- *featureCounts* (from *Subread* v1.6.2)
- *R* (version >= 3.0.0) with the following packages:
    - *cluster*
    - *edgeR*
    - *gplots*
    - *graphics*
    - *matrixStats*
    - *methods*
    - *RColorBrewer*
    - *stats*
    - *tools*

## Input Datasets

The input data files include:

- A set of compressed mRNA-sequencing *FASTQ* data files (**.fastq.gz*)
- The experiment design file `Conv-RNAseq-Experiment-Design.tsv.gz`
- The UCSC human genome reference library `Reference-Library.tar.gz`
- The data analysis programs `Analysis-Programs.tar.gz`
- The configuration file for the differential comparison analysis `DEG-Compare-Configs.tar.gz`
- The parameter files for the differential comparison analysis `DEG-Compare-Params.tar.gz`

## Analysis Workflow

**Step 1**

1. Create a direcotry tree with the following structure:

    ```
    Conv-GEO-Depot
    ├── Seqs
    ├── Aligns
    ├── Counts
    ├── References
    ├── Programs
    ├── Configs
    ├── Params
    ├── Scripts
    └── Results
    ```

2. Place all gzipped *FASTQ* data files in the `Seqs` directory. 

3. Extract the following compressed data files into corresponding directories:

    - Extract `Conv-RNAseq-Experiment-Design.tsv.gz` into the `Counts` directory.
    - Extract `Reference-Library.tar.gz` into the `References` directory.
    - Extract `Analysis-Programs.tar.gz` into the `Programs` directory.
    - Extract `DEG-Compare-Configs.tar.gz` into the `Configs` directory.
    - Extract `DEG-Compare-Params.tar.gz` into the `Params` directory.

4. Set the variable `DATASET_DIR` in all shell scripts (**.sh*) and the variable `dataset_dir` in the *R* program `Convert-Conv-RNAseq-Reads.GEO.R` under the `Programs` directory to the absolute path of the `Conv-GEO-Depot` directory.

**Step 2**

Build a set of index files for the human genome reference library used by *STAR*, by running `Programs/Sequence-Alignment/Build-STAR-Genome-Index.GEO.sh`.

**Step 3**

Align a set of mRNA-sequencing *FASTQ* data files to the human genome reference library, by running `Programs/Sequence-Alignment/Align-Conv-RNAseq-Reads.GEO.sh`.

**Step 4**

Count the number of aligned sequences for each reference gene, by running `Programs/Feature-Counts/Count-Conv-RNAseq-Reads.GEO.sh`.

**Step 5**

Convert the gene read-counts table for use with the DEG-comparison pipeline, by running `Rscript Programs/Feature-Counts/Convert-Conv-RNAseq-Reads.GEO.R`.

**Step 6**

Perform the differential comparison analysis on the gene read-counts table, by running `Run-Compare-Molecule-Expression.sh`.

## Output Results

- A table of uniquely aligned sequence reads for all reference genes `Conv-RNAseq-Read-Counts.tsv` in the `Counts` directory (compressed as `Conv-RNAseq-Read-Counts.tsv.gz`).
- A set of statistical comparison results for DEGs at each drug-treated condition `Human-[Cell Line]-Hour-48-Plate-[Plate Number]-Calc.CTRL-[Drug Name].tsv` in the `Results/FDR-[FDR Value]` directory (compressed as `DEG-Calc-FDR-0.1.tar.gz`).

